src/averageAges.js

```diff
@@ -37,7 +42,14 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => {
+    const someoneMothers = people.map(child => child.mother);
+
+    return person.sex === 'f' && (!withChildren
+      || someoneMothers.includes(person.name));
+  });
+
+  return getAverageAge(women.map(person => person.died - person.born));
```

Comment:
better to move the map logic to the constant - it improves readability of your code